pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -349,7 +349,7 @@ stdenv.mkDerivation ({
 
     env = stdenv.mkDerivation {
       name = "interactive-${pname}-${version}-environment";
-      nativeBuildInputs = [ ghcEnv systemBuildInputs ]
+      nativeBuildInputs = [ ghcEnv systemBuildInputs nativeBuildInputs ]
```

Comment:
I don’t quite follow. I personally use a generated `shell.nix` like:

```nix
((import <nixpkgs> {}).haskellPackages.override {
  overrides = self: super: {
        my-pkg = let
          buildDepends = with self; [
            ansi-wl-pprint
            trifecta
          ];
          in super.mkDerivation {
            pname = "pkg-env";
            src = "/dev/null";
            version = "none";
            license = "none";
            inherit buildDepends;
            buildTools = with self; [
              ghcid
              cabal-install
              hpack
              (hoogleLocal {
                packages = buildDepends;
              })
            ];
          };
      };
}).my-pkg
```

for my projects, where `buildTools` are the tools needed to build the package in the env.
Why wouldn’t a configured package work outside the shell environment as long as the paths are absolute?